src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,67 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final String EXISTING_USER_MESSAGE = "User with this login already exist";
+    private static final String INVALID_AGE = "Not valid age. Min allowed age: " + MIN_AGE;
+    private static final String INVALID_PASSWORD_LENGTH = "Not valid password length."
+            + " Min allowed password length is " + MIN_PASSWORD_LENGTH;
+    private static final String INVALID_LOGIN_LENGTH = "Not valid password length."
+            + " Min allowed password length is " + MIN_PASSWORD_LENGTH;
+    private static final String LOGIN_CANNOT_BE_NULL = "Login can't be null";
+    private static final String PASSWORD_CANNOT_BE_NULL = "Password can't be null";
+    private static final String AGE_CANNOT_BE_NULL = "Age can't be null";
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws RegistrationException {
+        try {
+            checkNullValues(user);
+            checkLength(user);
+            checkValidAge(user);
+            checkDB(user);
+        } catch (RegistrationException e) {
+            throw new RegistrationException();
+        }
+        return storageDao.add(user);
+    }
+
+    private boolean checkLength(User user) throws RegistrationException {
```

Comment:
Why did you use the boolean type in all private methods if you don't? Maybe void will be better?

Reply:
+, Boolean is redundant.